hashicorp/dev-portal
# feat: use next/image for docs images with known dimensions

## ✅ "Ready for Review" checklist




---

## 🔗 Relevant links



- [Preview link](https://dev-portal-git-BRANCH_NAME-hashicorp.vercel.app/PATH_TO_VIEW) 🔎
- [Asana task](url) 🎟️

## 🗒️ What

This PR removes our remark plugin that was previously used to rewrite image URLs to the Next.js image optimization endpoint in favor of using the native `next/image` element for image urls containing a `width` and `height` parameter (an indication of images being sourced from our content API static asset endpoint).

In order to support the `layout="raw"` option, this PR also updates Next to the next (lol) point release, `12.1.1`.

## 🤷 Why

Previously, we rewrote the URLs for images when rendering content, and manually appended the optimization endpoint URL. By using a native `next/image` element, we no longer need to maintain that logic, and are one step closer to enabling some of the more complex features like blurhashes.

## 🛠️ How

Parse the searchParams of `url` passed to our custom `Image` component, and if `width` and `height` are present, render a `next/image` instead of a `img`.

## 📸 Design Screenshots



## 🧪 Testing

- Go to preview URL:
- Confirm that the first image in the documentation renders correctly

## 💭 Anything else?
